update @mui/material dependencies to latest versions

There are some issues using this library with the latest version of @mui/material. Please investigate how we would update this package's dependencies.

@Jarodwr  

`The error: ✘ [ERROR] Missing "./_lib/format/longFormatters" specifier in "date-fns" package [plugin vite:dep-pre-bundle]`

I got this error because the scheduler was using date-fns v2.3 from the project instead of date-fns v4 from the library.
However, today the error not appeared. :|
But it could be because the library declares these packages in peerDependencies. When installing into the project, this may cause an error. So, I tried changing the packages from peerDependencies to dependencies, and the error disappeared.
P.s: I haven’t tried updating the library to the latest MUI for LIB yet.

--- a/package.json
+++ b/package.json
@@ -63,9 +63,6 @@
     "@eslint/compat": "^1.2.6",
     "@eslint/eslintrc": "^3.2.0",
     "@eslint/js": "^9.20.0",
-    "@mui/icons-material": ">=6.4.5 & <=6.4.12",
-    "@mui/material": ">=6.0.0 & <=6.4.12",
-    "@mui/x-date-pickers": ">=7.0.0 & <8.0.0",
     "@testing-library/dom": "^10.4.0",
     "@testing-library/jest-dom": "^6.6.3",
     "@testing-library/react": "^16.2.0",
@@ -76,7 +73,6 @@
     "@types/rollup-plugin-peer-deps-external": "^2",
     "@typescript-eslint/parser": "^8.24.1",
     "@vitejs/plugin-react": "^4.3.4",
-    "date-fns": ">=4.1.0",
     "eslint": "^9.20.1",
     "eslint-import-resolver-typescript": "^3.8.3",
     "eslint-plugin-import": "^2.31.0",
@@ -95,10 +91,9 @@
     "react-dom": "^19.0.0",
     "react-router": "^7.3.0",
     "rollup-plugin-peer-deps-external": "^2.2.4",
-    "rrule": "^2.8.1",
     "ts-jest": "^29.2.5",
     "ts-node": "^10.9.2",
-    "typescript": "^5.7.3",
+    "typescript": "^5.8.3",
     "typescript-eslint": "^8.24.1",
     "vite": "^6.1.1",
     "vite-plugin-checker": "^0.9.0",
@@ -106,14 +101,16 @@
     "vite-plugin-svgr": "^4.3.0",
     "vite-tsconfig-paths": "^5.1.4"
   },
+  "dependencies": {
+    "@mui/icons-material": "^7.0.0",
+    "@mui/material": "^7.0.0",
+    "@mui/x-date-pickers": "^7.0.0",
+    "date-fns": "^4.1.0",
+    "rrule": "^2.8.1"
+  },
   "peerDependencies": {
-    "@mui/icons-material": ">=6.4.5 & <=6.4.12",
-    "@mui/material": ">=6.0.0 & <=6.4.12",
-    "@mui/x-date-pickers": ">=7.0.0 & <8.0.0",
-    "date-fns": ">=4.0.0",
     "react": ">=18.0.0",
-    "react-dom": ">=18.0.0",
-    "rrule": ">=2.8.1"
+    "react-dom": ">=18.0.0"
   },
   "peerDependenciesMeta": {
     "rrule": {

--- a/src/lib/views/Editor.tsx
+++ b/src/lib/views/Editor.tsx
@@ -4,7 +4,7 @@ import {
   DialogActions,
   DialogContent,
   DialogTitle,
-  Grid2 as Grid,
+  Grid,
   useMediaQuery,
   useTheme,
 } from "@mui/material";

